lib/matplotlib/tests/test_axes.py

```diff
@@ -1763,6 +1765,9 @@ def test_scatter_color(self):
         ([0.5]*3, None),  # should emit a warning for user's eyes though
         ([0.5]*4, None),  # NB: no warning as matching size allows mapping
         ([0.5]*5, "shape"),
+        # list of strings:
+        (['0.5', '0.4', '0.6', '0.7'], None),
+        (['0.5', 'red', '0.6', 'C5'], None),
```

Comment:
Can you please add mixtures of string and float to the test? I.e. `c=[0.5, '0.5']` and `c=['0.5', 0.5]`.

Not sure what this should result in, but it should either be a defined output or raise an appropriate error. Either way it should be tested.

Reply:
Good idea.  Done!

Reply:
Note that it errors, which I think is correct.